test/cases/upgrade8to9.sh

```diff
@@ -0,0 +1,150 @@
+#!/bin/bash
+set -euxo pipefail
+
+# Get OS data.
+source /usr/libexec/osbuild-composer-test/set-env-variables.sh
+
+# Colorful output.
+function greenprint {
+    echo -e "\033[1;32m[$(date -Isecond)] ${1}\033[0m"
+}
+
+# Wait for VM to start
+function wait_for_vm {
+    INSTANCE_ADDRESS=192.168.100.50
+    SSH_OPTIONS=(-o StrictHostKeyChecking=no -o ConnectTimeout=5)
+    NUM_LOOPS=60
+    for _ in $(seq 0 "$NUM_LOOPS"); do
+        if eval 'sudo ssh "${SSH_OPTIONS[@]}" -q -i "${SSH_KEY}" root@"$INSTANCE_ADDRESS" exit'; then
+            echo "Test VM is up."
+            break
+        else
+            echo "Test VM is not ready yet."
+        fi
+        sleep 10
+    done
+}
+
+# Start libvirtd and test it.
+greenprint "🚀 Starting libvirt daemon"
+sudo systemctl start libvirtd
+sudo virsh list --all > /dev/null
+
+# Set a customized dnsmasq configuration for libvirt so we always get the
+# same address on bootup.
+sudo tee /tmp/integration.xml > /dev/null << EOF
+<network>
+  <name>integration</name>
+  <uuid>1c8fe98c-b53a-4ca4-bbdb-deb0f26b3579</uuid>
+  <forward mode='nat'>
+    <nat>
+      <port start='1024' end='65535'/>
+    </nat>
+  </forward>
+  <bridge name='integration' stp='on' delay='0'/>
+  <mac address='52:54:00:36:46:ef'/>
+  <ip address='192.168.100.1' netmask='255.255.255.0'>
+    <dhcp>
+      <range start='192.168.100.2' end='192.168.100.254'/>
+      <host mac='34:49:22:B0:83:30' name='vm' ip='192.168.100.50'/>
+    </dhcp>
+  </ip>
+</network>
+EOF
+if ! sudo virsh net-info integration > /dev/null 2>&1; then
+    sudo virsh net-define /tmp/integration.xml
+    sudo virsh net-start integration
+fi
+
+# Allow anyone in the wheel group to talk to libvirt.
+greenprint "🚪 Allowing users in wheel group to talk to libvirt"
+WHEEL_GROUP=wheel
+if [[ $ID == rhel ]]; then
+    WHEEL_GROUP=adm
+fi
+sudo tee /etc/polkit-1/rules.d/50-libvirt.rules > /dev/null << EOF
+polkit.addRule(function(action, subject) {
+    if (action.id == "org.libvirt.unix.manage" &&
+        subject.isInGroup("${WHEEL_GROUP}")) {
+            return polkit.Result.YES;
+    }
+});
+EOF
+
+# Prepare ssh key for the test VM
+SSH_DATA_DIR=$(/usr/libexec/osbuild-composer-test/gen-ssh.sh)
+SSH_KEY_PUB=${SSH_DATA_DIR}/id_rsa.pub
+SSH_KEY=${SSH_DATA_DIR}/id_rsa
+SSH_KEY_KS="sshkey --username root \"$(cat "$SSH_KEY_PUB")\""
+
+sudo tee ks.cfg > /dev/null << EOF
+text --non-interactive
```

Comment:
Is the initial installation and later upgrade progress logged into test file? It will be helpful to have them in case something fails.

Can you point me to a more precise place in the log output?

Reply:
No it's not logged anywhere. I'll output the console output somewhere and add it as job artifacts, that should show everything that's happening there.

Reply:
Logging into two separate files and uploading as artifacts now.